### ISSUE TITLE
Linux, recommended mono 4.0 - 4.2, how to install?

### ISSUE THREAD

**Issue by r3spawned** [ORelio/Minecraft-Console-Client#755]
Hi, 

I picked up a Raspberry Pi 3 b+, and have installed Raspbian (latest version).

I'm trying to install the recommended version of mono, which is any version between 4.0 - 4.2.

However, each time I try to install any older version of mono i find that version 4.6.2.7+dfsg-1 files gets installed, regardless of which snapshot i point to.

Does anybody have a set of commands i can type into terminal to force install an older version of mono?

Any version between 4-4.2 will do.

Thanks.

Reason : I'm having some issues with later mono versions, my player disconnects after typing certain commands, such as /prison (which I type to warp to a sub server).  When my player disconnects, nothing is displayed on screen, not even a "you have been disconnected", and this is even with debug mode turned on.  Its as if mono just dies.  I have no choice but to exit mono using ctrl+z.  I'm hoping an older version of mono may correct this.

**Comment by Mhowser** [comment]
I had no problem using the latest mono version on my raspberry pi 3 b+

**Comment by r3spawned** [comment]
The main issue I had was when exiting from a sub server to lobby then reentering the subserver, as a custom anti afk, but this only yielded in mono crashing.  So instead i am now using /connect <servername> followed by /warp <subserver) every 30 minutes.  Seems to have cleared up the issue.

**Comment by ORelio** [comment]
I think this bug is linked to workarounds implemented to have connection encryption while Mono does not implement the specific AES encryption mode Minecraft uses (CFB-8). We might be able to solve this once [.NET 5](https://devblogs.microsoft.com/dotnet/introducing-net-5/) is available since MCC will be able to run on .NET for Linux instead of Mono. Meanwhile, I'd recommend to add an AntiAFK command which does nothing such as /ping, that helps detecting when connection is lost.

**Comment by Sharpen-Up** [comment]
Sounds like the same issue as #569

**Comment by r3spawned** [comment]
Hi, as of yesterday I am seeing this message in terminal when I try to login to a subserver.  I havent changed anything so I'm not sure what has caused it to happen.

Im using SSH to connect via Putty (in Windows) to my raspberry pi.

I have tried installing raspbian version 10, to no avail.  The same error message appears.

I have reinstalled Mono, it is the latest stable version : 5.18.0.240+dsfg-3

This error appears sometimes when my player logs into a sub server.  It usually happens If I try to log in two different players in at the same time.   

Any ideas? Thanks.

__>
Unhandled Exception:
System.IO.InvalidDataException: Failed to process incoming packet of type KeepAlive. (PacketID: 0, Protocol: 47, LoginPhase: False, InnerException: System.IO.IOException). ---> System.IO.IOException: Unable to write data to the transport connection: The socket has been shut down. ---> System.Net.Sockets.SocketException: The socket has been shut down
  at System.Net.Sockets.Socket.Send (System.Byte[] buffer, System.Int32 offset, System.Int32 size, System.Net.Sockets.SocketFlags socketFlags) [0x0001a] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.Net.Sockets.NetworkStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 size) [0x000b4] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
   --- End of inner exception stack trace ---
  at System.Net.Sockets.NetworkStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 size) [0x00107] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at Org.BouncyCastle.Crypto.IO.CipherStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 count) [0x00029] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Crypto.Streams.MonoAesStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 count) [0x00000] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Protocol.Handlers.SocketWrapper.SendDataRAW (System.Byte[] buffer) [0x00008] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Protocol.Handlers.Protocol18Handler.SendPacket (System.Int32 packetID, System.Collections.Generic.IEnumerable`1[T] packetData) [0x000c1] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Protocol.Handlers.Protocol18Handler.SendPacket (MinecraftClient.Protocol.Handlers.PacketOutgoingType packet, System.Collections.Generic.IEnumerable`1[T] packetData) [0x0000d] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Protocol.Handlers.Protocol18Handler.HandlePacket (System.Int32 packetID, System.Collections.Generic.List`1[T] packetData) [0x00095] in <62ae6bad19544b3483b77e2c6c36bef5>:0
   --- End of inner exception stack trace ---
  at MinecraftClient.Protocol.Handlers.Protocol18Handler.HandlePacket (System.Int32 packetID, System.Collections.Generic.List`1[T] packetData) [0x00d55] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Protocol.Handlers.Protocol18Handler.Update () [0x00036] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Protocol.Handlers.Protocol18Handler.Updater () [0x00007] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00017] in <d0e12f672b88444ab4b6d9b2ecf20142>:0
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x0008d] in <d0e12f672b88444ab4b6d9b2ecf20142>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <d0e12f672b88444ab4b6d9b2ecf20142>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x00031] in <d0e12f672b88444ab4b6d9b2ecf20142>:0
  at System.Threading.ThreadHelper.ThreadStart () [0x0000b] in <d0e12f672b88444ab4b6d9b2ecf20142>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.IO.InvalidDataException: Failed to process incoming packet of type KeepAlive. (PacketID: 0, Protocol: 47, LoginPhase: False, InnerException: System.IO.IOException). ---> System.IO.IOException: Unable to write data to the transport connection: The socket has been shut down. ---> System.Net.Sockets.SocketException: The socket has been shut down
  at System.Net.Sockets.Socket.Send (System.Byte[] buffer, System.Int32 offset, System.Int32 size, System.Net.Sockets.SocketFlags socketFlags) [0x0001a] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.Net.Sockets.NetworkStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 size) [0x000b4] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
   --- End of inner exception stack trace ---
  at System.Net.Sockets.NetworkStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 size) [0x00107] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at Org.BouncyCastle.Crypto.IO.CipherStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 count) [0x00029] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Crypto.Streams.MonoAesStream.Write (System.Byte[] buffer, System.Int32 offset, System.Int32 count) [0x00000] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Protocol.Handlers.SocketWrapper.SendDataRAW (System.Byte[] buffer) [0x00008] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Protocol.Handlers.Protocol18Handler.SendPacket (System.Int32 packetID, System.Collections.Generic.IEnumerable`1[T] packetData) [0x000c1] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Protocol.Handlers.Protocol18Handler.SendPacket (MinecraftClient.Protocol.Handlers.PacketOutgoingType packet, System.Collections.Generic.IEnumerable`1[T] packetData) [0x0000d] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Protocol.Handlers.Protocol18Handler.HandlePacket (System.Int32 packetID, System.Collections.Generic.List`1[T] packetData) [0x00095] in <62ae6bad19544b3483b77e2c6c36bef5>:0
   --- End of inner exception stack trace ---
  at MinecraftClient.Protocol.Handlers.Protocol18Handler.HandlePacket (System.Int32 packetID, System.Collections.Generic.List`1[T] packetData) [0x00d55] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Protocol.Handlers.Protocol18Handler.Update () [0x00036] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at MinecraftClient.Protocol.Handlers.Protocol18Handler.Updater () [0x00007] in <62ae6bad19544b3483b77e2c6c36bef5>:0
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00017] in <d0e12f672b88444ab4b6d9b2ecf20142>:0
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x0008d] in <d0e12f672b88444ab4b6d9b2ecf20142>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <d0e12f672b88444ab4b6d9b2ecf20142>:0
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x00031] in <d0e12f672b88444ab4b6d9b2ecf20142>:0
  at System.Threading.ThreadHelper.ThreadStart () [0x0000b] in <d0e12f672b88444ab4b6d9b2ecf20142>:0__
